### ISSUE TITLE
stops drawing when navigating with react router dom 

### ISSUE THREAD

**Issue by zilahir** [Gherciu/react-p5#25]
I have a Router, and a component which implements a `<Sketch />`. This is placed on the top of the Router, so it will stay there on every page. When i am pragmatically navigating between the pages, the p5 drawing _stops_. The canvas is still exists in the DOM, but it gets cleared. Only way to get them back is by refreshing the entire page. 

Any idea how to go around this problem? 

Thank you!

**Comment by Gherciu** [comment]
Hi, @zilahir  can you please provide a minimal example on (http://codesandbox.io/)[http://codesandbox.io/].
Thanks!

**Comment by zilahir** [comment]
Hi, @Gherciu! 

I did a quick demo project, but a basic route, and its not happening there, so i think the problem in my own implementation. 

This is the [repo ](https://github.com/zilahir/difference) for it. 

The implementatino of the sketch is [here](https://github.com/zilahir/difference/blob/master/src/components/common/Background/index.jsx). 

Thanks for taking a look! :)

**Comment by zilahir** [comment]
UPDATE: 

funny thing, but it seems the issue is not with the `Routing`. If i change the drawing function to the example you provided in this repo for example, the problem is gone, so i think i have managed to narrow it done to the images the p5 rendering....

**Comment by zilahir** [comment]
UPDATE: 

I have managed to solve the problem. 

The images I was loading had been moved to `Rect.useState([])`, as well as as other `p5` related stuff such as `coordinates`. All fo the setup had been moved to the stup, and where `Array` was needed, I used the `length` of the array that had filled up in the `preload()` function. Such as number of images, etc. 

After doing all this, the issue does not exists anymore, so it was not a problem reagrding this package, but an implementation issue.
